### PR TITLE
create functions for work with secrets

### PR DESCRIPTION
**Pull request by delgod**
This PR aims to move secret implementation logic inside the separate functions.
As a result, we can change the implementation of storing/fetching secrets later without touching the logic of charm itself.

NB: inside the code, we have two types of secrets.
Some secrets are shared between units, and some should be accessible and unique only for a particular unit.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -232,6 +231,39 @@ def app_data(self) -> Dict:
 
         return relation.data[self.app]
 
+    @property
+    def unit_data(self) -> Dict:
```
Comment (jnsgruk) [inline src/charm.py:235]: I'd probably name this `unit_peer_data` or similar

**Review by jnsgruk — APPROVED** [review]
I like the idea - should make migrating to Juju Secrets easier. Modulo tests, I'm +1

File: src/charm.py (hunk 1 above)
Comment (delgod) [reply src/charm.py:235]: thanks, I will create a separate PR for renaming

**Review by MiaAltieri — APPROVED** [review]
🤩

**Pull request merged**